packages/react-components/src/components/TextFieldWithMention/mentionTagHelpers.ts

```diff
@@ -0,0 +1,957 @@
+    };
+  }
+  let result = '';
+  let plainTextSelectionEndIndex: number | undefined = undefined;
+  let rangeStart: number;
+  let rangeEnd: number;
+  // check if space symbol is handled in case if string looks like '<1 2 3>'
+  let isSpaceLengthHandled = false;
+  rangeStart = oldPlainText.lastIndexOf(' ', startIndex);
+  if (rangeStart !== -1 && rangeStart !== undefined && rangeStart > tag.plainTextBeginIndex) {
+    isSpaceLengthHandled = true;
+  }
+  rangeEnd = oldPlainText.indexOf(' ', oldPlainTextEndIndex);
+  if (rangeEnd === -1 || rangeEnd === undefined) {
+    // check if space symbol is not found
+    rangeEnd = plainTextEndIndex;
+  } else if (!isSpaceLengthHandled) {
+    // +1 to include the space symbol
+    rangeEnd += 1;
+  }
+  isSpaceLengthHandled = true;
+
+  if (rangeStart === -1 || rangeStart === undefined || rangeStart < tag.plainTextBeginIndex) {
+    // rangeStart should be at least equal to tag.plainTextBeginIndex
+    rangeStart = tag.plainTextBeginIndex;
+  }
+  if (rangeEnd > plainTextEndIndex) {
+    // rangeEnd should be at most equal to plainTextEndIndex
+    rangeEnd = plainTextEndIndex;
+  }
+  if (rangeStart === tag.plainTextBeginIndex && rangeEnd === plainTextEndIndex) {
+    // the whole tag should be removed
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex) + processedChange;
+    plainTextSelectionEndIndex = tag.plainTextBeginIndex + processedChange.length;
+    processedChange = '';
+    lastProcessedHTMLIndex = closeTagIndex + closeTagLength;
+  } else {
+    // only part of the tag should be removed
+    let startChangeDiff = 0;
+    let endChangeDiff = 0;
+    // need to check only rangeStart > tag.plainTextBeginIndex as when rangeStart === tag.plainTextBeginIndex startChangeDiff = 0 and mentionTagLength shouldn't be subtracted
+    if (rangeStart > tag.plainTextBeginIndex) {
+      startChangeDiff = rangeStart - tag.plainTextBeginIndex - mentionTagLength;
+    }
+    endChangeDiff = rangeEnd - tag.plainTextBeginIndex - mentionTagLength;
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex + tag.openTagBody.length + startChangeDiff);
+
+    if (startIndex < tag.plainTextBeginIndex) {
+      // if the change is before the tag, the selection should start from startIndex (rangeStart will be equal to tag.plainTextBeginIndex)
+      plainTextSelectionEndIndex = startIndex + change.length;
+    } else {
+      // if the change is inside the tag, the selection should start with rangeStart
+      plainTextSelectionEndIndex = rangeStart + processedChange.length;
+    }
+    lastProcessedHTMLIndex = tag.openTagIndex + tag.openTagBody.length + endChangeDiff;
+    // processed change should not be changed as it should be added after the tag
+  }
+  return { result, updatedChange: processedChange, htmlIndex: lastProcessedHTMLIndex, plainTextSelectionEndIndex };
+};
+
+/**
+ * Closing tag information
+ *
+ * @private
+ */
+export type ClosingTagInfoResult = {
```

Comment:
Why name this way? 
`ClosingTagInfo` is better maybe?

Reply:
@emlynmac I asked Viktoriia the same question in the previous PR. It is because ClosingTagInfoResult is the result type for `getTagClosingTagInfo`.
However, since this is also only used explicitly in one place, I'll define it inline with the method parameters.